r/R/csv.R

```diff
@@ -418,6 +418,8 @@ CsvTableReader$create <- function(file,
 #' The `CsvWriteOptions$create()` factory method takes the following arguments:
 #' - `include_header` Whether to write an initial header line with column names
 #' - `batch_size` Maximum number of rows processed at a time. Default is 1024.
+#' - `null_string` The string to be written for null values. Must not contain
+#'   quotation marks. Default is `NA`.
```

Comment:
Is this the current default, or is this a behavior change?

Reply:
For both `write_csv_arrow` and `write_dataset` it was empty string by default. I was originally thinking of aligning with readr default of "NA", at least in `write_csv_arrow`. But now I'm not so sure. @thisisnic what do you think?

Reply:
Hmm, I'm not sure. It'd be nice to have at least `write_csv_arrow()` behaviour aligned with `readr`, but this would be a change in behaviour that users might not want if they're used to the current behaviour.  What do you think @nealrichardson?

Reply:
Perhaps our goal should be to support the `readr` options, but not necessarily to match the `readr` defaults. Does that seem like a reasonable policy? Right now the function takes `readr` arguments as a convenience, and also has Arrow-specific parameter that may overlap. It seems like if we wanted an API that matched the defaults, we should have a separate API that exclusively took `readr` options.

Reply:
I think that's a reasonable compromise, I don't think we should be changing default values in this PR just to match readr, but would be good to support the parameter, so that works.